front/src/Components/Glossarium.tsx

```diff
@@ -0,0 +1,253 @@
+        this.setState({
+          selectedText: selection.toString(),
+          evaluateInputText: selection.toString()
+        });
+      }
+    }
+  }
+
+  selectionEnd = () => {
+    document.removeEventListener('mouseup', this.selectionEnd);
+    this.setState({
+      selectingText: false
+    });
+  }
+
+  componentDidMount() {
+    document.addEventListener('selectionchange', this.selectionChangeCallback);
+
+    const pdfViewer = document.getElementById("PDFViewer");
+    if (pdfViewer) {
+      pdfViewer.addEventListener('selectstart', () => {
+        this.setState({
+          selectingText: true
+        });
+        document.addEventListener('mouseup', this.selectionEnd);
+      });
+    }
+
+    const glossaryDescription = document.getElementById("glossary_item_definition");
+    if (glossaryDescription) {
+      glossaryDescription.addEventListener('selectstart', () => {
+        this.setState({
+          selectingText: true
+        });
+        document.addEventListener('mouseup', this.selectionEnd);
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    const pdfViewer = document.getElementById("PDFViewer");
+    if (pdfViewer) {
+      pdfViewer.removeEventListener('selectionchange', this.selectionChangeCallback);
+    }
+    const glossaryDescription = document.getElementById("glossary_item_definition");
+    if (glossaryDescription) {
+      glossaryDescription.removeEventListener('selectionchange', this.selectionChangeCallback);
+    }
+    document.removeEventListener('selectionchange', this.selectionChangeCallback);
+  }
+
+
+  render() {
+    return (
+      <div className="Glossarium">
+        <div className="glossarium-container">
+          <div className="selection-container">
+            <h1>Huidige selectie</h1>
+            <form onSubmit={this.evaluateSelection}>
+              <input
+                className="selected-text"
+                value={this.state.evaluateInputText}
+                onChange={this.onChange}/>
+              <Button className="evaluate-selection-button">
+                  Verzenden
+              </Button>
+            </form>
+          </div>
+
+          {!this.state.loading && <div className="definition-container">
+            <div className="definition-title">
+              {this.state.customTopic == true && <b>{this.state.topicCanonicalName} + ({this.state.topicAbbreviation})</b>}
+            </div>
+            <div className="definition" id="glossary_item_definition">
+              {this.state.customTopic == true && <a href={this.state.topicSource} target="_blank" rel="noopener noreferrer">Bron</a>}
+              <p>{this.state.topicDescription}</p>
+            </div>
+          </div>}
+
+          {this.state.loading && <div className="definition-container">Zoeken...</div>}
+
+          <div className="linked-data-container">
+            <div className="wiki-summary">
+              {this.state.foundOnWikipedia == true && <p>{this.state.information}</p>}
```

Comment:
prefer ternary operator

```typescript
this.state.foundOnWikipedia ? <p>thing</p> : <p>otherthing</p>
```

Reply:
Don't seem to be able to use ternary operator here as i have no "else" part to my if statement.